front/src/common/components/screen-reader-snackbar/screen-reader-snackbar.hook.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { ScreenReaderSnackbarOptions } from './screeen-reader-snackbar.vm';
+import { ScreenReaderSnackbarContext } from './screen-reader-snackbar.context';
+
+export const useScreenReaderSnackbarContext = () => {
+  const { options, setOptions } = React.useContext(ScreenReaderSnackbarContext);
+
+  const optionsRef = React.useRef<ScreenReaderSnackbarOptions>({
+    messages: [],
+  });
+
+  const [intervalScreen, setIntervalScreen] = React.useState(null);
+
+  return {
+    showScreeanReaderMessage: (message: string) => {
+      optionsRef.current = {
+        messages: [...optionsRef.current.messages, message],
+      };
+      setOptions(optionsRef.current);
+      const interval = setInterval(() => {
```

Comment:
Why do you need use setTimeout and setInterval?, Why don't use only setTimeout? something like:

```
export const useScreenReaderSnackbarContext = () => {
  const { options, setOptions } = React.useContext(ScreenReaderSnackbarContext);
  const optionsRef = React.useRef<ScreenReaderSnackbarOptions>(options);

  const timer = React.useRef(null);

  React.useEffect(() => {
    return () => {
      console.log('Se ha limpiado', timer.current);
      clearTimeout(timer.current);
    };
  }, []);

  const showScreenReaderMessage = (message: string) => {
    optionsRef.current = {
      messages: [...optionsRef.current.messages, message],
    };
    setOptions(optionsRef.current);
    timer.current = setTimeout(() => {
      optionsRef.current = {
        messages: optionsRef.current.messages.filter(
          (m,) => m !== message
        ),
      };
      setOptions(optionsRef.current);
    }, options.timeout);
  };

  return {
    showScreenReaderMessage,
  };
};
```